core/src/queue.rs

```diff
@@ -209,75 +207,67 @@ impl Queue {
                 continue;
             }
 
-            seen.push(hash);
             if *entry
                 .get()
                 .check_signature_condition(wsv)
                 .expect("Checked in `check_tx` just above")
             {
-                return Some(entry.get().clone());
-            }
-        }
-    }
-
-    /// Pop a single transaction.
-    ///
-    /// Unlike [`Self::pop`], unsigned transactions are not recorded.
-    #[allow(
-        clippy::expect_used,
-        clippy::unwrap_in_result,
-        clippy::cognitive_complexity
-    )]
-    pub fn pop_without_seen(&self, wsv: &WorldStateView) -> Option<VersionedAcceptedTransaction> {
```

Comment:
This method was root cause of the issue.
Because this methods doesn't remove `(hash,tx)` record from the `queue.txs` and doesn't put back seen transactions to be deleted later, when `tx` would be in the blockchain.